cppcache/include/geode/Exception.hpp

```diff
@@ -42,87 +44,49 @@ class StackTrace;
  * @class Exception Exception.hpp
  * A description of an exception that occurred during a cache operation.
  */
-class CPPCACHE_EXPORT Exception {
+class CPPCACHE_EXPORT Exception : public std::exception {
   /**
    * @brief public methods
    */
  public:
   /** Creates an exception.
    * @param  msg1 message pointer, this is copied into the exception.
-   * @param  msg2 optional extra message pointer, appended to msg1.
-   * @param  forceTrace enables a stacktrace for this exception regardless of
-   * stacktrace-enabled system property.
-   * @param  cause optional cause of the exception which can be later
-   *               retrieved using <code>getCause</code>
    **/
-  Exception(const char* msg1, const char* msg2 = nullptr,
-            bool forceTrace = false, const std::shared_ptr<Exception>& cause = nullptr);
```

Comment:
Concerned about the loss of `cause`. Can you provide rational for dropping it?

Reply:
Primary rational was that it isn't part of `std::exception`.  I'll try playing around with `std::nested_exception`.